test/env/data.go

```diff
@@ -179,3 +184,79 @@ func BundleHasSyncedAllNamespaces(ctx context.Context, cl client.Client, name, e
 
 	return true
 }
+
+// CheckBundleSynced returns nil if the given Bundle has synced the expected data to the given
+// namespace, or else returns a descriptive error if that's not the case.
+// - Skips Namespaces that are Terminating since targets are not synced there.
+// - Ensures the Bundle status has been updated with the appropriate target.
+// - Ensures the Bundle has the correct status condition with the same ObservedGeneration as the current Generation.
+func CheckBundleSynced(ctx context.Context, cl client.Client, bundleName string, namespace string, expectedData string) error {
```

Comment:
This function looks a LOT like `BundleHasSynced` are there any differences, can we merge the two functions?

Reply:
Difference is that it returns an error so that failures can be passed to callers in the normal way. Means that if the `CheckBundleSynced` fails when called inside an `Eventually` and it times out, you see where it went wrong.

Totally fair that it's very similar; I'll remove `BundleHasSynced` and any uses of it.